packages/fxa-auth-server/lib/routes/subscriptions/stripe-webhook.ts

```diff
@@ -490,12 +488,40 @@ export class StripeWebhookHandler extends StripeHandler {
       return;
     }
     const invoice = event.data.object as Stripe.Invoice;
+    const paypalInvoice = await this.stripeHelper.invoicePayableWithPaypal(
+      invoice
+    );
+
+    // Fetch the customer to determine if we need to set their name.
+    // This is needed because the original PayPal implementation did not copy
+    // the name to the Stripe customer object. This is a workaround for that
+    // issue, with a metric to track the impact and indicate when this can be
+    // removed.
+    const customer = await this.stripeHelper.expandResource(
+      invoice.customer,
+      CUSTOMER_RESOURCE
+    );
     if (
-      !(await this.stripeHelper.invoicePayableWithPaypal(invoice)) ||
-      invoice.status !== 'draft'
+      paypalInvoice &&
+      customer &&
+      customer.deleted !== true &&
+      !customer.name
     ) {
+      const billingAgreementId =
+        this.stripeHelper.getCustomerPaypalAgreement(customer);
+      // The customer needs to be updated for their name.
+      if (billingAgreementId) {
+        await this.paypalHelper.updateStripeNameFromBA(
+          customer,
+          billingAgreementId
+        );
+      }
+    }
+
+    if (!paypalInvoice || invoice.status !== 'draft') {
       return;
     }
```

Comment:
I'm surprised this isn't up before we call `expandResource` on the `customer`. Could this be moved up accordingly to save some work/time?